chains/ethereum/listener.go

```diff
@@ -87,7 +93,13 @@ func (l *Listener) buildQuery(contract ethcommon.Address, sig EventSig) eth.Filt
 // Handler will be called for every instance of event.
 func (l *Listener) RegisterEventHandler(subscription string, handler chains.EvtHandlerFn) error {
 	evt := EventSig(subscription)
-	query := l.buildQuery(l.cfg.emitter, evt)
+	log15.Trace("Event signatures", "sig", evt, "topic", evt.GetTopic().Hex())
+	var query eth.FilterQuery
+	if subscription == DepositProposalSignature {
+		query = l.buildQuery(l.cfg.receiver, evt)
+	} else {
+		query = l.buildQuery(l.cfg.emitter, evt)
+	}
```

Comment:
This is a bandaid, we should handle this higher in the stack. #140 changes how we call `RegisterEventHandler`, perhaps we can also specify the contract as part of the new subscription struct

Reply:
Yep! Although @spacesailor24 is introducing a pr that brings us down to one contract that we interact with. I can add a TODO to remove this or wait for @spacesailor24 to submit his PR and branch off that

Reply:
Fair point, please add a todo